std/machines/large_field/rotate.asm

```diff
@@ -0,0 +1,66 @@
+use std::utils::unchanged_until;
+use std::utils::cross_product;
+use std::convert::int;
+
+// Rotate for single bytes using an exhaustive table
+// TODO this way, we cannot prove anything that rotates by more than 31 bits.
+machine ByteRotate with
+    latch: latch,
+    operation_id: operation_id,
+    degree: 65536
+{
+    operation run<0> P_operation, P_A, P_B, P_ROW -> P_C;
+
+    col fixed latch = [1]*;
+    col fixed operation_id = [0]*;
+
+    let bit_counts = [256, 32, 4, 2];
+    let min_degree = std::array::product(bit_counts);
+    std::check::assert(std::prover::min_degree() >= std::array::product(bit_counts), || "The rotate machine needs at least 65536 rows to work.");
+    let inputs = cross_product(bit_counts);
+    let a: int -> int = inputs[0];
+    let b: int -> int = inputs[1];
+    let row: int -> int = inputs[2];
```

Comment:
I think we should find a better name for this. Maybe 'out_byte_offset'?

Reply:
yes, it would be better